pkg/server/server.go

```diff
@@ -72,6 +75,41 @@ type KiamServer struct {
 	parallelFetchers    int
 }
 
+// GetPodCredentials returns credentials for the Pod, according to the role it's
+// annotated with. It will additionally check policy before returning credentials.
+func (k *KiamServer) GetPodCredentials(ctx context.Context, req *pb.GetPodCredentialsRequest) (*pb.Credentials, error) {
+	pod, err := k.pods.GetPodByIP(req.Ip)
+	if err != nil {
+		if err == k8s.ErrPodNotFound {
+			return nil, ErrPodNotFound
+		}
+
+		return nil, err
+	}
+	logger := log.WithFields(k8s.PodFields(pod)).WithField("pod.iam.requestedRole", req.Role)
+
+	decision, err := k.assumePolicy.IsAllowedAssumeRole(ctx, req.Role, req.Ip)
+	if err != nil {
+		logger.Errorf("error checking policy: %s", err.Error())
+		return nil, err
+	}
+
+	if !decision.IsAllowed() {
+		logger.WithField("policy.explanation", decision.Explanation()).Errorf("pod denied by policy")
+		return nil, ErrPolicyForbidden
+	}
+
+	creds, err := k.credentialsProvider.CredentialsForRole(ctx, req.Role)
```

Comment:
So another potentially breaking change will come here when we want to specify the session name by pod ID. I don't know if that is important enough to roll in here or wait until later. I think the usefulness of a constant session name is fairly low.

Reply:
There's a separate issue open for being able to name sessions #38. 

The signature for `CredentialsForRole` will change but the only published API is the gRPC server's `GetPodCredentials` which should tolerate changes to reference pods by IP/ID etc.

So for now I'd leave this and pick it up as part of #38.